bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlFactory.java

```diff
@@ -97,10 +108,12 @@ public static BXml parse(InputStream xmlStream) {
         try {
             XmlTreeBuilder treeBuilder = new XmlTreeBuilder(new InputStreamReader(xmlStream));
             return treeBuilder.parse();
+        } catch (BError e) {
+            throw e;
         } catch (DeferredParsingException e) {
             throw ErrorCreator.createError(StringUtils.fromString((e.getCause().getMessage())));
         } catch (Throwable e) {
-            throw ErrorCreator.createError(StringUtils.fromString(("failed to create xml: " + e.getMessage())));
+            throw ErrorCreator.createError(StringUtils.fromString((PARSE_ERROR_PREFIX + e.getMessage())));
```

Comment:
Do we need '(' here

Reply:
Will remove this.

Reply:
Fixed with eb82f105a413cf5fec654f770a2802c197c35fa2.